readme.md

```diff
@@ -119,6 +119,38 @@ The `wikimedia/server` config consists of `wikimedia`, `wikimedia/node` and `wik
 }
 ```
 
+### TypeScript
+TypeScript cannot be directly handled by the browser (or ResourceLoader), so it will always target either code that runs
+on the server or tests or code that is compiled to JavaScript first in a non-MediaWiki step. Adjust the outer "extends"
+accordingly. If you have a "mixed" project or are adopting TypeScript incrementally then you can use the typescript
+config only for TypeScript files like so:
+
+`.eslintrc.json`:
+```json
+{
+	"root": true,
+	"extends": [
+		"wikimedia/server"
+	],
+	"overrides": [
```

Comment:
These overrides should be in common.json so that all .ts files are linted with this config automatically.

Reply:
This will automatically opt in all typescript files (unless the eslint command is restricted to only some file-types).

But looking at the repos that use TypeScript (https://codesearch.wmcloud.org/things/?q=import&files=.*%5C.ts%24&excludeFiles=&repos=), this does not seem too bad. Wikibase is already using this config, and so are Growth repositories (GrowthExperiments, CommunityConfiguration, GuidedTours).

I'll move the overrides as suggested.